src/Microsoft.AspNetCore.Routing/Patterns/RoutePatternFactory.cs

```diff
@@ -257,18 +292,28 @@ public static RoutePattern Pattern(string rawText, params RoutePatternPathSegmen
             // It's important that these two views of the data are consistent. We don't want
             // values specified out of line to have a different behavior.
 
-            var updatedDefaults = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase);
-            if (defaults != null)
+            Dictionary<string, object> updatedDefaults = null;
+            if (defaults != null && defaults.Count > 0)
             {
+                if (updatedDefaults == null)
+                {
+                    updatedDefaults = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase);
```

Comment:
More defaults could be added later from a part.

Reply:
Not necessarily. With a `List<T>` the default internal capacity is 4. If you initialize the capacity to 1, then add 2 items, the list will resize.

Anyway, if I have time I'll change it and see whether it makes a positive or negative change to the results.

Reply:
It didn't make any difference, but that is because none of the endpoints in the benchmark have defaults or policies. I looked at `Dictionary<K, V>`'s source and setting the capacity will never make perf worse so I'll include it.